hy/cmdline.py

```diff
@@ -58,6 +58,9 @@ def __init__(self, spy=False, output_fn=None, locals=None,
 
         self.spy = spy
 
+        # Queue size of stored REPL results
+        self._repl_results_queue_len = 3
```

Comment:
Why not, instead of going through a range of numbers and redoing the mangling every time, do it once ahead of time?

```python
self._repl_results_vars = [mangle("*{}".format(i) for i in range(3))]
```

Then its just a mater of going through the list:

```python
value = ...
for var in self._repl_results_vars:
    self.locals[var], value = value, self.locals[var]
```

Reply:
I was thinking about this, after checking the behavior of python. It seems that `_` will be undefined if no result from previous lines: Accessing `_` throws error.

I didn't check iPython but assume it might be the same. I guess we would like keep such behavior for `*x`, what do you think?

Reply:
I don't think that's necessary, or even particularly desirable.

Reply:
@Kodiologist - What the default value then? It might be a little weird If we use `None` as the default value, as we don't allow None appears in `*x`.

The implementation could be much simpler if we don't want this behavior, just initialize and shift (I don't even think the queue size need to be configurable):

Reply:
@waigx you sure?

```
ipython
Python 3.6.4 (default, Jan  5 2018, 02:35:40) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: _
Out[1]: ''
```

Reply:
```
I didn't check iPython but assume it might be the same.
```

hahaha 😄

Reply:
alright, you guys win 😤

Reply:
What the initial values then - apparently iPython is using `''`

Reply:
`None` or `''` works for me.

Reply:
no problem at all - I just used `None` as default since that makes more sense to me. The code has been updated (literally copied you suggestion above). Please take another look.